R/install.R

```diff
@@ -396,3 +398,172 @@ build_status_ok <- function(process_log, quiet = FALSE) {
+          )
+          stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+        } else {
+          if (!quiet) message("Writing RTools path to ~/.Renviron ...")
+          write('PATH="${RTOOLS40_HOME}\\usr\\bin;${RTOOLS40_HOME}\\mingw64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          stop("Restart R and run check_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.", call. = FALSE)
+        }
+      }
+      # Check if the mingw32-make and g++ get picked up by default are the RTools-supplied ones
+      if (toolchain_path != mingw32_make_path || gpp_path != toolchain_path) {
+        if (!fix) {
+          error_message <- c(
+            "",
+            "Other C++ toolchains installed on your system conflict with RTools.",
+            "Run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)' to fix the issue."
+          )
+          stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+        } else {
+          if (!quiet) message("Writing RTools path to ~/.Renviron ...")
+          write('PATH="${RTOOLS40_HOME}\\usr\\bin;${RTOOLS40_HOME}\\mingw64\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)
+          stop("Restart R and run check_cmdstan_toolchain(check_only = TRUE) to confirm the installation was successful.", call. = FALSE)
+        }
+      }
+    } else {
+      mingw32_make_path <- dirname(Sys.which("mingw32-make"))
+      gpp_path <- dirname(Sys.which("g++"))
+      # If mingw32-make and g++ are not found, we check typical RTools 3.5 folders.
+      # If found, we fix PATH, otherwise we recommend the user to install RTools 3.5.
+      if (!nzchar(mingw32_make_path) || !nzchar(gpp_path)) {
+        rtools_path <- Sys.getenv("RTOOLS35_HOME")
+        if (!nzchar(rtools_path)) {
+          typical_paths <- c(
+            file.path("C:/", "Rtools"),
+            file.path("C:/", "Rtools35"),
+            file.path("C:/", "RTools"),
+            file.path("C:/", "RTools35"),
+            file.path("C:/", "rtools"),
+            file.path("C:/", "rtools35"),
+            file.path("c:/", "Rtools"),
+            file.path("c:/", "Rtools35"),
+            file.path("c:/", "RTools"),
+            file.path("c:/", "RTools35"),
+            file.path("c:/", "rtools"),
+            file.path("c:/", "rtools35")
+          )
+          for (p in typical_paths) {
+            if (dir.exists(p)) {
+              rtools_path <- p
+            }
+          }
+        }
+        if (nzchar(rtools_path)) {
+          if (!fix) {
+            error_message <- c(
+              "",
+              "A RTools installation was found, but the PATH is not properly set.",
+              "Run 'cmdstanr::check_cmdstan_toolchain(check_only = FALSE)' to fix the issue."
+            )
+            stop(paste0(error_message, collapse = "\n"), call. = FALSE)
+          }
+          if (!quiet) message("Writing RTools path to ~/.Renviron ...")
```

Comment:
Nitpicking for style consistency: 

```suggestion
          if (!quiet) {
            message("Writing RTools path to ~/.Renviron ...")
          }
```